sklearn/pipeline.py

```diff
@@ -416,7 +455,10 @@ def fit_predict(self, X, y=None, **fit_params):
         y_pred : array-like
         """
         Xt, fit_params = self._fit(X, y, **fit_params)
-        return self.steps[-1][-1].fit_predict(Xt, y, **fit_params)
+        with _print_elapsed_time('Pipeline',
+                                 self._log_message(len(self.steps) - 1)):
```

Comment:
The message will indicate that fitting took x seconds, when in reality we're reporting the fit + predict time. Fitting took less than x seconds.

Same for `fit_transform`.

I would suggest passing a `method` parameter to `_log_message` so that it can accommodate:

- (step %d of %d) Fitting ...
- (step %d of %d) Fit-predicting ...
- (step %d of %d) Fit-transforming ...